docs/pages/developer-guide.md

```diff
@@ -936,6 +936,160 @@ In the example above, a server task can be accessed through the address:
+
+The TLS artifacts within single task will be shared across all instances of the pod task. Different tasks can request TLS artficats with same name but each task will get unique private key and certificate and they won't get shared across different tasks.
+
+In a given example the `$MESOS_SANDBOX` directory would contain following files:
+
+```
+$MESOS_SANDBOX/
+               ...
+               tls.key
+               tls.crt
+               tls.ca
+               ...
+```
+
+## Provisioning
+
+TLS artfits are provisioned by a **scheduler** based on the service configuration. Generated artifacts are stored as secrets in the `default` secrets store. The scheduler stores each artifact (private key, certificate, CA bundle, keystore and truststore) as a separate secret under the tasks `DCOS_SPACE` path. This approach ensures that tasks launched by scheduler [will get access](operations-guide.html#authorization-for-secrets) to all necessary secrets. If all secrets exist for single artifact then it is **not** overwritten and existing value is used. Currently there is no exposed automated way of regenerating TLS artifacts. Operator can delete secrets from DC/OS secret store which will trigger generating new TLS artifacts.
+
+The scheduler will generate and store TLS artfiacts for both possible formats (`TLS`, `KEYSTORE`) and changing format won't result in new private key.
+
+Generated artifacts are stored as secrets with following naming scheme:
+
+```
+[DCOS_SPACE]/[pod-index-taskname]__[transport-encryption-name]__[artifact-type]
+```
+
+Example of secrets that would get created based on YAML configuration above:
+
+```
+hello-world/hello-0-server__tls__certificate
+hello-world/hello-0-server__tls__private-key
+hello-world/hello-0-server__tls__root-ca-certificate
+hello-world/hello-0-server__tls__keystore
+hello-world/hello-0-server__tls__truststore
+```
+
+### TLS artifacts secrets lifetime
+
+When `transport-encryption` item is removed from the service configuration (i.e. YAML file) it is **not** removed from the secret store. If the same `transport-encryption` configuration is added back to the service configuration existing TLS artifact will be used.
+
+The `UninstallScheduler` responsible for cleaning up service installation will try to remove all TLS artifact secrets previously provisioned by the scheduler.
+
+### Task artifacts access
+
+Each task that requests one or more TLS artfiacts will get artifacts delivered to the task sandbox directory (`$MESOS_SANDBOX`) as a file-based secret. This feature ensures that data aren't stored on disk and are held only in memory on virtual filesystem. Secret appears as a normal file that can be used in application. A task will get only TLS artifacts that are configured in service configuration.
+
+## Private key and certificate details
+
+A scheduler provisions *private key* and `X.509` *certificate* and exposed them to task in various encodings.
```

Comment:
- "exposes" instead of "exposed"
- "serialization formats" instead of "encodings" (that's the term that is typically used in libraries that want to abstract across various serialization formats)